tests/observability-test-utils/src/main/java/org/ballerina/testobserve/listenerendpoint/Resource.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerina.testobserve.listenerendpoint;
+
+import io.ballerina.runtime.api.types.ResourceFunctionType;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.values.BObject;
+
+import java.util.Objects;
+
+import static io.ballerina.runtime.api.constants.TypeConstants.STRING_TNAME;
+import static org.ballerina.testobserve.listenerendpoint.Constants.CALLER_TYPE_NAME;
+import static org.ballerina.testobserve.listenerendpoint.Constants.TEST_OBSERVE_PACKAGE;
+
+/**
+ * Holds information related to a Ballerina service resource.
+ */
+public class Resource {
+    private final BObject serviceObject;
+    private final ResourceFunctionType resourceFunctionType;
+    private final String basePath;
+
+    public Resource(BObject serviceObject, ResourceFunctionType resourceFunctionType, String basePath) {
+        this.serviceObject = serviceObject;
+        this.resourceFunctionType = resourceFunctionType;
+        this.basePath = basePath;
+
+        // Validate resource
+        int paramCount = this.getParamTypes().length;
+        String resourcePath = this.getResourcePath();
+        if (paramCount > 2) {
```

Comment:
What does this validation do? Is this specific to the test case?

Reply:
This is a validation for the mock listener. The listener at the moment, only supports resource functions and only accepts two arguments.

The validation is there as a way to throw an error if a developer doesn't adhere to the expected format.